### PR TITLE
chore(tiered): minor fixes + expose buffer allocation type stats

### PR DESCRIPTION
**Pull request by romange**
Also fixes small_bins deletion logic.
Fixes https://github.com/dragonflydb/dragonfly/issues/3240

### CODE REVIEW

**Review by dranikpg — APPROVED** [review]

File: src/server/tiering/small_bins_test.cc [hunk 1]
```diff
@@ -95,4 +100,18 @@ TEST(SmallBins, PartialStashDelete) {
   }
 }
 
+TEST_F(SmallBinsTest, Bug3240) {
```
Comment (dranikpg) [inline src/server/tiering/small_bins_test.cc:103]: Maybe give it a proper name describing the scenarion and then keep a comment that is was a bug. I mean testing for fixed bugs is not worth it, but testing scenarios (not covered by tests) surely is.

Comment (romange) [reply src/server/tiering/small_bins_test.cc]: sure

File: src/server/tiering/small_bins.cc [hunk 2]
```diff
@@ -67,18 +82,22 @@ SmallBins::FilledBin SmallBins::FlushBin() {
   }
 
   current_bin_bytes_ = 0;
-  current_bin_.erase(current_bin_.begin(), current_bin_.end());
+  current_bin_.clear();
```
Comment (dranikpg) [inline src/server/tiering/small_bins.cc:85]: https://github.com/abseil/abseil-cpp/blob/master/absl/container/flat_hash_set.h#L226-L228

**Review by dranikpg — APPROVED** [review]

File: src/server/tiering/small_bins.cc (hunk 2 above)
Comment (romange) [reply src/server/tiering/small_bins.cc]: added comment

**Review by dranikpg — APPROVED** [review]

**Pull request merged**